mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulSettingsScreen.kt

```diff
@@ -38,13 +40,17 @@ class AuthenticatedUserProfileScreenState(
   override val puzzlesCount = 0
   private val onEditProfileNameClickAction by onEditProfileNameClickAction
   private val onEditProfileImageClickAction by onEditProfileImageClickAction
+  private val onLogoutAction by onLogoutAction
 
   override fun onEditProfileNameClick() {
     onEditProfileNameClickAction()
   }
   override fun onEditProfileImageClick() {
     onEditProfileImageClickAction()
   }
+  override fun onLogout() {
+    onLogoutAction()
```

Comment:
I think you should not pass an action as a parameter, but implement it here instead. Make `scope` a `private val` if needed.

```suggestion
    scope.launch { user.signOut() }
```